responsive_image/templatetags/responsive_image_tags.py

```diff
@@ -85,9 +92,15 @@ def render(self, context):
             canonical_url = obj.image.url
             name, extension = canonical_url.rsplit(".", 1)
 
-            for di, size_name in zip(srcsets, size_names.split(",")):
-                url = getattr(obj, "get_%s_url" % size_name)()
-                di["url"] = url
+            try:
+                for di, size_name in zip(srcsets, size_names.split(",")):
+                    url = getattr(obj, "get_%s_url" % size_name)()
+                    di["url"] = url
+            except AttributeError:
```

Comment:
hm ok cool. I also didn't like this solution but couldn't find any info online as to a better approach will investigate thanks :)